packages/react/src/components/Tooltip/index.tsx

```diff
@@ -17,6 +17,23 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
   hideElementOnHidden?: boolean;
 }
 
+// fires a custom "cauldron:tooltip:show" / "cauldron:tooltip:hide" event
+// to allow projects using cauldron to hook into when a tooltip is shown/hidden
+const fireCustomEvent = (show: boolean, button?: HTMLElement | null) => {
+  if (!button) {
+    return;
+  }
+
+  const event = new Event(
```

Comment:
LGTM! I wonder if this should (eventually) be an event factory.

Something like `new CauldronEvents('tooltip:show', ...details)`. We can wait to refactor when implementing #162